openshift/dynamic/client.py

```diff
@@ -66,18 +90,44 @@ class DynamicClient(object):
         the kubernetes API
     """
 
-    def __init__(self, client):
+    def __init__(self, client, cache_file=None):
         self.client = client
         self.configuration = client.configuration
+        default_cache_id = self.configuration.host
+        if PY3:
+            default_cache_id = default_cache_id.encode('utf-8')
+        self.__resources = ResourceContainer({}, client=self)
+        self.__cache_file = cache_file or '/tmp/osrcp-{0}.json'.format(base64.b64encode(default_cache_id).decode('utf-8'))
```

Comment:
should probably use `$TMPDIR` rather than defaulting to `/tmp`

Reply:
Good catch